spinalcordtoolbox/scripts/sct_maths.py

```diff
@@ -270,223 +315,465 @@ def __call__(self, _, namespace, values, option_string=None):
-        if dim + 1 > len(np.shape(data)):  # in case input volume is 3d and dim=t
-            data = data[..., np.newaxis]
-        data_out = np.mean(data, dim)
-
-    elif arg_name == "rms":
-        dim = dim_list.index(arg_value)
-        if dim + 1 > len(np.shape(data)):  # in case input volume is 3d and dim=t
-            data = data[..., np.newaxis]
-        data_out = np.sqrt(np.mean(np.square(data.astype(float)), dim))
-
-    elif arg_name == "std":
-        dim = dim_list.index(arg_value)
-        if dim + 1 > len(np.shape(data)):  # in case input volume is 3d and dim=t
-            data = data[..., np.newaxis]
-        data_out = np.std(data, dim, ddof=1)
-
-    elif arg_name == "smooth":
-        sigmas = arg_value
-        if len(sigmas) == 1:
-            sigmas = [sigmas[0] for i in range(len(data.shape))]
-        elif len(sigmas) != len(data.shape):
-            printv(parser.error(
-                'ERROR: -smooth need the same number of inputs as the number of image dimension OR only one input'))
-        # adjust sigma based on voxel size
-        sigmas = [sigmas[i] / dim[i + 4] for i in range(3)]
-        # smooth data
-        data_out = sct_math.smooth(data, sigmas)
-
-    elif arg_name == "dilate":
-        if arg_value[1] in ['disk', 'square'] and arg_value[2] is None:
-            printv(parser.error('ERROR: -dim is required for -dilate with 2D morphological kernel'))
-        data_out = sct_math.dilate(data, size=arg_value[0], shape=arg_value[1], dim=arg_value[2])
-
-    elif arg_name == "erode":
-        if arg_value[1] in ['disk', 'square'] and arg_value[2] is None:
-            printv(parser.error('ERROR: -dim is required for -erode with 2D morphological kernel'))
-        data_out = sct_math.erode(data, size=arg_value[0], shape=arg_value[1], dim=arg_value[2])
-
-    elif arg_name == "denoise":
-        # parse denoising arguments
-        p, b = 1, 5  # default arguments
-        list_denoise = arg_value.split(",")
-        for i in list_denoise:
-            if 'p' in i:
-                p = int(i.split('=')[1])
-            if 'b' in i:
-                b = int(i.split('=')[1])
-        data_out = sct_math.denoise_nlmeans(data, patch_radius=p, block_radius=b)
+            values.append(np.full(shape, arg))
+    return header, values
+
 
+def apply_add(current_value, *args):
+    """Implementation for -add."""
+    # special case to operate on a 4D volume across the t axis
+    if isinstance(current_value, Image) and current_value.data.ndim == 4 and not args:
+        return Image(
+            np.sum(current_value.data, axis=3),
+            hdr=current_value.hdr,
+        )
+
+    header, values = get_header_and_values(current_value, args)
+    if header is None:
+        return sum(values)
     else:
-        assert arg_name == "symmetrize"
-        data_out = sct_math.symmetrize(data, arg_value)
+        return Image(
+            np.sum(values, axis=0),
+            hdr=header,
+        )
```

Comment:
Note that the `Image` constructor automatically adjusts the header to match the shape of the data array.